src/WebAppResolver.ts

```diff
@@ -17,10 +18,17 @@ export class WebAppResolver implements AppResourceResolver {
 
             if (this.siteCacheLastUpdated < Date.now() - 1000 * 3) {
                 this.siteCacheLastUpdated = Date.now();
+
                 this.listWebAppsTask = new Promise((resolve, reject) => {
                     this.siteCache.clear();
+                    this.siteNameCounter.clear();
+
                     uiUtils.listAllIterator(client.webApps.list()).then((sites) => {
                         for (const site of sites) {
+                            const siteName: string = nonNullProp(site, 'name');
```

Comment:
Adds a lookahead and counts duplicate names and adds location descriptions to help differentiate app names to users, example:

![image](https://github.com/user-attachments/assets/26d07935-e0ed-413f-936e-eef22dcd6df4)